hikari/auto_mod.py

```diff
@@ -0,0 +1,230 @@
+
+    MESSAGE_SEND = 1
+    """When a member sends or edits a message in the guild."""
+
+
+class AutoModTriggerType(int, enums.Enum):
+    """Type of trigger for an auto-moderation rule."""
+
+    KEYWORD = 1
+    """Match message content against a list of keywords."""
+
+    HARMFUL_LINK = 2
+    """Scan messages for links which are deemed "harmful" by Discord."""
+
+    SPAM = 3
+    """Discord's guild anti-spam system."""
+
+    KEYWORD_PRESET = 4
+    """Discord's preset keyword triggers."""
+
+
+class AutoModKeywordPresetType(int, enums.Enum):
+    """Discord's KEYWORD_PRESET type."""
+
+    PROFANITY = 1
+    """Trigger on words which may be considered forms of swearing or cursing."""
+
+    SEXUAL_CONTENT = 2
+    """Trigger on words that refer to explicit behaviour or activity."""
+
+    SLURS = 3
+    """Trigger on personal insults and words which "may be considered hate speech"."""
+
+
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class PartialAutoModTrigger:
+    """Base class representing the content a rule triggers on."""
+
+    type: AutoModTriggerType = attr.field(eq=False, hash=False, repr=False)
+    """The type action this triggers."""
+
+
+@attr.define(kw_only=True, weakref_slot=False)
+class KeywordTrigger(PartialAutoModTrigger):
+    """A trigger based on matching message content against a list of keywords."""
+
+    keyword_filter: typing.Sequence[str] = attr.field(eq=False, hash=False, repr=False)
+    """The filter strings this trigger checks for.
+
+    This supports a wildcard matching strategy which is documented at
+    https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies.
+    """
+
+
+class HarmfulLinkTrigger(PartialAutoModTrigger):
+    """A trigger based on Discord's own list of links deemed "harmful"."""
+
+    __slots__: typing.Sequence[str] = []
+
+
+class SpamTrigger(PartialAutoModTrigger):
+    """A trigger based on Discord's spam detection."""
+
+    __slots__: typing.Sequence[str] = []
+
+
+@attr.define(kw_only=True, weakref_slot=False)
+class KeywordPresetTrigger(PartialAutoModTrigger):
+    """A trigger based on a predefined set of presets provided by Discord."""
+
+    allow_list: typing.Sequence[str] = attr.field(eq=False, factory=list, hash=False, repr=False)
+    """A sequence of filters which will be exempt from triggering the preset trigger.
+
+    This supports a wildcard matching strategy which is documented at
+    https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies.
+    """
+
+    presets: typing.Sequence[typing.Union[int, AutoModKeywordPresetType]] = attr.field(eq=False, hash=False, repr=False)
+    """The predefined presets provided by Discord to match against."""
+
+
+@attr_extensions.with_copy
+@attr.define(hash=True, kw_only=True, weakref_slot=False)
+class AutoModRule(snowflakes.Unique):
```

Comment:
Commenting here, but not sure if it's the right place. Discord recently added a new field to add your custom message to display to the user

https://github.com/discord/discord-api-docs/commit/111e0bd6e7f87fb0a9bb3c6f04a745dea8b09549